eth2near/utilities/src/eth_proof_generator.rs

```diff
@@ -0,0 +1,178 @@
+use crate::{
+    eth_rpc_client::EthRPCClient,
+    primitives::*,
+    types::{BlockHeader, Log, TransactionReceipt},
+};
+use cita_trie::{MemoryDB, PatriciaTrie, Trie, TrieError};
+use hasher::HasherKeccak;
+use rlp::RlpStream;
+use std::error::Error;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct Proof {
+    pub log_index: u64,
+    pub log_entry_data: Vec<u8>,
+    pub receipt_index: u64,
+    pub receipt_data: Vec<u8>,
+    pub header_data: Vec<u8>,
+    pub proof: Vec<Vec<u8>>,
+}
+
+pub fn get_proof_for_event(tx_hash: &str, log_index: u64, node_url: &str,) -> Result<Proof, Box<dyn Error>> {
+    let client = EthRPCClient::new(node_url);
+
+    let tx_bytes: [u8; 32] = hex::decode(&tx_hash[2..])
+        .expect("Invalid hex string")
```

Comment:
Let's return an error instead of panic here

Reply:
Fixed in c70cc73eb4cd78f8dfdfb2f1be34a6807cc5c921